.github/workflows/build-and-publish-images.yml

```diff
@@ -0,0 +1,143 @@
+name: Build and publish images workflow
+
+on:
+  workflow_call:
+    inputs:
+      repo:
+        type: string
+      branch:
+        type: string
+      date:
+        type: string
+      sha:
+        type: string
+      build_type:
+        required: true
+        type: string
+
+      push:
+        required: true
+        type: boolean
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: none
+  checks: none
+  contents: read
+  deployments: none
+  discussions: none
+  id-token: write
+  issues: none
+  packages: read
+  pages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Compute matrix
+        id: compute-matrix
+        run: ./ci/compute-matrix.sh
+  docker-build-per-arch:
+    needs: compute-matrix
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.MATRIX) }}
+    runs-on: ${{ fromJSON(matrix.runner_arch != 'arm64' && '"ubuntu-latest"' || '{"labels":["self-hosted", "linux", "arm64", "cpu4"]}') }}
```

Comment:
We need to change `'{"labels":["self-hosted", "linux", "arm64", "cpu4"]}'` into `'linux-arm64-cpu4'` to use ARC V2 runners

Reply:
Updated!